From a9a121a0aa0fde095099db03651d83aaf0cbcbc5 Mon Sep 17 00:00:00 2001
From: Jose Fernandez <josef@netflix.com>
Date: Mon, 20 May 2024 21:57:28 -0600
Subject: [PATCH] Do not error out if journald is not available

When journald is not available, the program should not error out, but
instead make noop log entries.

Signed-off-by: Jose Fernandez <josef@netflix.com>
---
 src/main.rs | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/main.rs b/src/main.rs
index 00fc067..d95a4c1 100755
--- a/src/main.rs
+++ b/src/main.rs
@@ -118,9 +118,15 @@ fn main() -> Result<()> {
         return Err(anyhow!("This program must be run as root"));
     }
 
+    // Initialize the journald layer or ignore if not available
+    let journald_layer = match tracing_journald::layer() {
+        Ok(layer) => Some(layer),
+        Err(_) => None,
+    };
+
     // Initialize the tracing subscriber with the journald layer
     let registry = tracing_subscriber::registry()
-        .with(tracing_journald::layer()?)
+        .with(journald_layer)
         .with(tracing_subscriber::filter::LevelFilter::INFO);
     // Try to set this subscriber as the global default
     registry.try_init()?;
